test/compile/scale.test.ts

```diff
@@ -258,6 +258,86 @@ describe('Scale', function() {
     });
   });
 
+  describe('ordinal with color', function() {
+    const encoding = parseModel({
+      mark: "point",
+      encoding: {
+        color: { field: 'origin', type: "ordinal"}
+      }
+    });
+
+    const scales = vlscale.compileScales([COLOR], encoding);
+
+    it('should create color and inverse scales', function() {
+      assert.equal(scales.length, 2);
+      assert.equal(scales[0].name, 'inverse_rank');
+      assert.equal(scales[1].name, 'color');
+    });
+
+    it('should create correct inverse scale', function() {
+      assert.equal(scales[0].type, 'ordinal');
+      assert.deepEqual(scales[0].domain, {
+        data: 'source',
+        field: 'rank_origin',
+        sort: true
+      });
+      assert.deepEqual(scales[0].range, {
+        data: 'source',
+        field: 'origin',
+        sort: true
+      });
+    });
+
+    it('should create correct color scale', function() {
+      assert.equal(scales[1].type, 'linear');
+      assert.deepEqual(scales[1].domain, {
+        data: 'source',
+        field: 'rank_origin'
+      });
+    });
+  });
+
+  describe('color with bin', function() {
+    const encoding = parseModel({
+        mark: "point",
+        encoding: {
+          color: { field: 'origin', type: "quantitative", bin: true}
+        }
+      });
+
+    const scales = vlscale.compileScales([COLOR], encoding);
+
+    it('should add correct scales', function() {
+      assert.equal(scales.length, 3);
+
+      assert.equal(scales[0].name, 'inverse_rank');
+      assert.equal(scales[1].name, 'color_label');
+      assert.equal(scales[2].name, 'color');
+    });
+
+    it('should create correct inverse scale', function() {
+      assert.equal(scales[0].type, 'ordinal');
+      assert.deepEqual(scales[0].domain, {
+        data: 'source',
+        field: 'bin_origin_start',
+        sort: true
+      });
+      assert.deepEqual(scales[0].range, {
+        data: 'source',
+        field: 'bin_origin_start',
+        sort: true
+      });
+    });
+
+    it('should sort range of color labels', function() {
+      assert.deepEqual(scales[1].range, {
+        data: 'source',
+        field: 'bin_origin_range',
+        sort: {"field": "bin_origin_start","op": "min"}
+      });
+    });
+  });
+
```

Comment:
Missing test for timeUnit.   I'd say add least let's add test scaffolding, so later someone can add additional test for it.